### PR TITLE
minor: Update data type support documentation

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

N/A

# Rationale for this change

https://arrow.apache.org/datafusion/user-guide/sql/data_types.html is out of date, as we discovered while working on docs for IOx: https://github.com/influxdata/docs-v2/pull/4700

# What changes are included in this PR?
1. Update supported type
2. Add note about `arrow_typeof` function

# Are these changes tested?

N/A (they are docs)

# Are there any user-facing changes?

Yes, docs

### CODE REVIEW

File: docs/source/user-guide/sql/data_types.md [hunk 1]
```diff
@@ -25,13 +25,26 @@ execution. The SQL types from
...
+```
+
 ## Character Types
 
 | SQL DataType | Arrow DataType |
 | ------------ | -------------- |
 | `CHAR`       | `Utf8`         |
 | `VARCHAR`    | `Utf8`         |
 | `TEXT`       | `Utf8`         |
+| `STRING`     | `Utf8`         |
```
Comment (alamb) [inline docs/source/user-guide/sql/data_types.md:47]:
```sql
❯ create table foo (x string) as select 'foo';
0 rows in set. Query took 0.003 seconds.
```

File: .github/workflows/dev.yml [hunk 2]
```diff
@@ -42,8 +42,7 @@ jobs:
           node-version: "14"
       - name: Prettier check
         run: |
-          # if you encounter error, try rerun the command below with --write instead of --check
-          # and commit the changes
+          # if you encounter error, rerun the command below and commit the changes
```
Comment (alamb) [inline .github/workflows/dev.yml:45]: driveby cleanup (the command below actually uses `--write` rather than `--check` already

**Review by andygrove — APPROVED** [review]

**Review by xudong963 — APPROVED** [review]
